extensions/skyportal/static/js/components/CentroidPlot.jsx

```diff
@@ -0,0 +1,495 @@
+import React, { useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import PropTypes from "prop-types";
+import { makeStyles, useTheme } from "@material-ui/core/styles";
+import embed from "vega-embed";
+import * as d3 from "d3";
+import convertLength from "convert-css-length";
+import * as photometryActions from "../ducks/photometry";
+import * as archiveActions from "../ducks/archive";
+
+const useStyles = makeStyles(() => ({
+  centroidPlotDiv: (props) => ({
+    width: props.width,
+    height: props.height,
+  }),
+}));
+
+// Helper functions for computing plot points (taken from GROWTH marshall)
+const gcirc = (ra1, dec1, ra2, dec2) => {
+  // input deg, haversine formula, return deg
+  ra1 = (ra1 / 180) * Math.PI;
+  dec1 = (dec1 / 180) * Math.PI;
+  ra2 = (ra2 / 180) * Math.PI;
+  dec2 = (dec2 / 180) * Math.PI;
+  const delDec2 = (dec2 - dec1) * 0.5;
+  const delRA2 = (ra2 - ra1) * 0.5;
+  return (
+    (360 *
+      Math.asin(
+        Math.sqrt(
+          Math.sin(delDec2) * Math.sin(delDec2) +
+            Math.cos(dec1) *
+              Math.cos(dec2) *
+              Math.sin(delRA2) *
+              Math.sin(delRA2)
+        )
+      )) /
+    Math.PI
+  );
+};
+
+const relativeCoord = (ra, dec, refRA, refDec) => {
+  const delRA = gcirc(ra, dec, refRA, dec) * 3600 * -Math.sign(ra - refRA);
+  const delDec = gcirc(ra, dec, ra, refDec) * 3600 * Math.sign(dec - refDec);
+  return { delRA, delDec };
+};
+
+// Temporary function for draft
+const getReferencePoint = (ras, decs) => {
+  const refRA = d3.median(ras);
+  const refDec = d3.median(decs);
+
+  return { refRA, refDec };
+};
```

Comment:
I vaguely remember that this wasn't originally what the GROWTH marshal was doing in terms of reference point for the offsets, to just take the median? I think this was what we implemented since we didn't have the crossmatch info before. Do we still want the reference to just be the median of the data instead of the nearest crossmatch or whatever it was for the GROWTH marshal?

Reply:
To my taste, this makes the most sense: the reference point will be calculated using all ZTF photometry (which is usually a lot of points) and then we'll show all the cross-matched sources from other catalogs within 10" (could be many) relative to that.

Reply:
Sure makes sense to me

Reply:
There's an implicit assumption here that the uncertainties are the same for all the photometry points. Should consider a weighted average (after sigma clipping).

Also, how will this snippet behave if there are any nans, none, etc. in `ras`, `decs`

Reply:
@profjsb Regarding nans/nones: we preselect only the photometry data points with non-null RA/Decl, see https://github.com/fritz-marshal/fritz/pull/306/files#diff-3608c51f5312a907615fed7fcbc9f46b0cff6700b05032f7f51a58d21aa6517cR341. 

As for the uncertainties, ZTF alerts contain SExtractor'ed parameters of a PSF-fit, such as the FWHM (see https://github.com/ZwickyTransientFacility/ztf-avro-alert/blob/master/schema/candidate.avsc#L37), which are not converted to uncertainties in RA/Dec's. We don't do that on the K side, and only post RA/Dec's (see https://github.com/dmitryduev/kowalski/blob/master/kowalski/alert_broker_ztf.py#L1165). We probably should, but I think it's outside of the scope of this PR.